bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/utils/Utils.java

```diff
@@ -34,6 +35,16 @@
  */
 @NonNullByDefault
 public class Utils {
+    private static Clock clock = Clock.systemDefaultZone();
```

Comment:
Since static, you will be locked into the same time-zone even if the system time-zone will change. Also, openHAB time-zone is not considered (it could be different than system time-zone).

Reply:
It might not matter, depending on how this `Clock` object is used.

Reply:
Okay, after another look I can see that the handler calls setClock with openHAB's time-zone, so this is only a default `Clock` object in case `setClock` is not called before `getClock`.

Reply:
Above code assures `clock` is not null at the beginning and avoid null checks in the code afterwards.

`SolarForecastHandlerFactory` immediately calls in constructor `setClock` to assure `clock` is according to configured time-zone. No other handler is delivered during this time, right?